### PR TITLE
Describe the Design of the Helpful Engineering AWS Organization and Accounts

### PR DESCRIPTION
**Pull request by skuenzli**
This is a first pass at documenting a/the design for the Helpful Engineering AWS Organization and Account structure.  

We've discussed much of this in Slack, implemented some if it, and I'd like to document our approach and open it up to wider comment.

In a future revision, this or another doc should describe:

* SSO for HelpfulEng team members
* IAM principals for project delivery and service operations

### CODE REVIEW

File: documentation/aws-account-and-identity-architecture.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# AWS Account and Identity Architecture
+
+This document describes the Helpful Engineering account and identity architecture for Amazon Web Services (AWS).
+
+## Problem Statement
+
+[Helpful Engineering](https://helpfulengineering.org) is creating and deploying applications to AWS as part of 
+its mission to address the COVID-19 pandemic.  These applications are needed or developed by project teams
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
its mission to address the COVID-19 pandemic. These applications are needed or developed by project teams 
```

File: documentation/aws-account-and-identity-architecture.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+This document describes the Helpful Engineering account and identity architecture for Amazon Web Services (AWS).
+
+## Problem Statement
+
+[Helpful Engineering](https://helpfulengineering.org) is creating and deploying applications to AWS as part of 
+its mission to address the COVID-19 pandemic.  These applications are needed or developed by project teams 
+staffed by volunteers from all over the world.
+
+The applications and project teams develop and operate independently, have limited time, and use a wide set of 
+implementation technologies.  HelpfulEng has neither the time nor the staff to support high touch, high coordination
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
implementation technologies. Helpful Engineering has neither the time nor the staff to support high touch, high coordination
```

File: documentation/aws-account-and-identity-architecture.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+## Solution Benefits
+
+Helpful Engineering needs an AWS account and identity architecture that maximizes data safety, team autonomy, and
+execution efficiency.
+
+The solution should:
+
+* enable project delivery teams to deploy applications safely and independently
+* safeguard customer and organization data from external and internal threats, both accidental and malicious
+* minimize demand for scarce DevOps & Cloud skills
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
* Enable project delivery teams to deploy applications safely and independently.
* Safeguard customer and organization data from external and internal threats, both accidental and malicious.
* Minimize demand for scarce DevOps & Cloud skills.
```

File: documentation/aws-account-and-identity-architecture.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+## Key Use Cases
+
+### UC1 - Deliver & Operate Applications Independently
+
+HelpfulEng project teams and deployments are independent, do not generally depend on each other, and expect a high 
+degree of autonomy.  Project teams need to be able to develop applications, deliver them, and perform some amount of
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
degree of autonomy. Project teams need to be able to develop applications, deliver them, and perform some amount of 
```

File: documentation/aws-account-and-identity-architecture.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+## Key Use Cases
+
+### UC1 - Deliver & Operate Applications Independently
+
+HelpfulEng project teams and deployments are independent, do not generally depend on each other, and expect a high 
+degree of autonomy.  Project teams need to be able to develop applications, deliver them, and perform some amount of 
+operational activities in a self-service, low coordination manner.
+
+We anticipate that project teams will use a mix of manual and automated Cloud configuration and application deployment.
+Manual configurations are likely to be used in early prototyping and development.  The DevOps team will enable and
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
Manual configurations are likely to be used in early prototyping and development. The DevOps team will enable and 
```

File: documentation/aws-account-and-identity-architecture.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+degree of autonomy.  Project teams need to be able to develop applications, deliver them, and perform some amount of 
+operational activities in a self-service, low coordination manner.
+
+We anticipate that project teams will use a mix of manual and automated Cloud configuration and application deployment.
+Manual configurations are likely to be used in early prototyping and development.  The DevOps team will enable and 
+encourage automated configurations delivery and operation to production. The DevOps team has no plans to enforce that.
+
+### UC2 - Provision Accounts
+
+The HelpfulEng DevOps team will provision AWS accounts for both shared and project delivery accounts.  The DevOps team
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
The HelpfulEng DevOps team will provision AWS accounts for both shared and project delivery accounts. The DevOps team
```

File: documentation/aws-account-and-identity-architecture.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+would like provision these accounts in a standardized way with low effort and simple adoption of Cloud security and
+governance practices.
+
+## Logical Architecture
+
+This design provisions a set of AWS accounts for each project team to deliver their applications and a few shared
+accounts for prototyping, security, and governance.  The Helpful Engineering AWS Organization will look like:
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
accounts for prototyping, security, and governance. The Helpful Engineering AWS Organization will look like: 
```

File: documentation/aws-account-and-identity-architecture.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  └ Monitoring O2
+    └ he-project-monitoring-O2-dev
+    └ he-project-monitoring-O2-prod
+  └ <project name>
+    └ he-<project name>-dev
+    └ he-<project name>-prod
+  └ ...
+```
+
+The design accommodates the wide set of people, skills, and techniques used throughout HelpfulEng and scopes the
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
The design accommodates the wide set of people, skills, and techniques used throughout Helpful Engineering and scopes the
```

File: documentation/aws-account-and-identity-architecture.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+management, fault, and security domain to the project team.  Using separate AWS accounts for each project team 
+creates a strong security boundary between teams that isolates each project's activities, resources, and
+data from each other.  
+
+The Helpful Engineering AWS Organization OU will be configured to match the depicted hierarchy.  This OU
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
The Helpful Engineering AWS Organization OU will be configured to match the depicted hierarchy. This OU 
```

File: documentation/aws-account-and-identity-architecture.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+The design accommodates the wide set of people, skills, and techniques used throughout HelpfulEng and scopes the
+management, fault, and security domain to the project team.  Using separate AWS accounts for each project team 
+creates a strong security boundary between teams that isolates each project's activities, resources, and
+data from each other.  
+
+The Helpful Engineering AWS Organization OU will be configured to match the depicted hierarchy.  This OU 
+structure enables use of both shared and project-specific Service Control Policy.
+
+The DevOps team will use [AWS Control Tower](https://docs.aws.amazon.com/controltower/latest/userguide/what-is-control-tower.html) 
+to provision both the shared and project accounts.  Control Tower provides a simple account provisioning model that
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
to provision both the shared and project accounts. Control Tower provides a simple account provisioning model that
```

File: documentation/aws-account-and-identity-architecture.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+provides a number of security and governance best practices out of the box. 
+
+**Core Accounts**
+
+The `Security` account contains the organization's security logs such as CloudTrail logs.
+
+The `Audit` account contains the organization's audit findings such as Config and GuardDuty reports.
+
+** Sandbox Accounts **
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
**Sandbox Accounts**
```

File: documentation/aws-account-and-identity-architecture.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+The `he-sandbox2` account is a '[sandbox](https://chariotsolutions.com/blog/post/building-developer-sandboxes-on-aws/)' 
+account teams can use to experiment and prototype solutions.  Applications should not be operated for customers out of 
+sandbox  accounts.  The DevOps team should consider enforcing this policy with a governance tool like Cloud Custodian
+that destroys resources some number of days after provisioning, e.g. 10 days.
+
+** Project Delivery Accounts**
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
**Project Delivery Accounts**
```

File: documentation/aws-account-and-identity-architecture.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+The DevOps team will provision two 'delivery' accounts for each project team: `dev` and `prod`.  Teams will use the
+`dev` account to develop their applications and test application deployments.  Applications should be delivered
+to the production account for operation and use by customers and end users.  
+
+The project team is responsible for delivering applications to their accounts with support by the DevOps team.  
+
+Project teams should adopt automated continuous integration to build application artifacts.  These artifacts can be
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
Project teams should adopt automated continuous integration to build application artifacts. These artifacts can be 
```

File: documentation/aws-account-and-identity-architecture.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+`dev` account to develop their applications and test application deployments.  Applications should be delivered
+to the production account for operation and use by customers and end users.  
+
+The project team is responsible for delivering applications to their accounts with support by the DevOps team.  
+
+Project teams should adopt automated continuous integration to build application artifacts.  These artifacts can be 
+stored in:
+ 
+* a trusted external repository such as Docker Hub 
+* an internal repository such as an S3 bucket hosted within a project account as is the case for the Serverless Framework
```
Comment (0x2b3bfa0) [inline documentation/aws-account-and-identity-architecture.md]:
```suggestion
* A trusted external repository such as Docker Hub.
* An internal repository such as an S3 bucket hosted within a project account as is the case for the Serverless Framework.
```

**Comment by rsareth** [comment]
Will you recommend to set up a CI/CD (workflows and tools)? You introduced the **dev** and **prod** environments.

Rigth now, with the current urgency, I guess some developers are pushing on the **master** branch, run the UTs on their machine, etc. Maybe some of them have setup their own CI/CD on any SaaS. I think at some point we should ask them to provide the command lines(s) to package and/or to deploy their projects. It would make our tasks easier to deploy everything.

**Review by imonthercks — APPROVED** [review]

File: documentation/aws-account-and-identity-architecture.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+
+## Logical Architecture
+
+This design provisions a set of AWS accounts for each project team to deliver their applications and a few shared
+accounts for prototyping, security, and governance. The Helpful Engineering AWS Organization will look like: 
+
+```
+HelpfulEng AWS Org
+└ Core
+  └ Security
```
Comment (chris-mac) [inline documentation/aws-account-and-identity-architecture.md]: Default Control Tower accounts are log and audit

**Review by chris-mac — APPROVED** [review]

File: documentation/aws-account-and-identity-architecture.md (hunk 15 above)
Comment (skuenzli) [reply documentation/aws-account-and-identity-architecture.md]: Thx @chris-mac - I updated the account name and descriptions to match AWS CT docs.

**Pull request merged**